src/main/java/com/uk/mybatisdemo202401/MovieController.java

```diff
@@ -17,13 +17,18 @@ public List<Movie> findAll() {
         return movieService.findAll();
     }
 
-    @PostMapping("/nameList")
-    public List<Movie> findByNames(@RequestParam String movieName) {
-        return movieMapper.findByNameContains(movieName);
+    @GetMapping("/movies/{id}")
+    public Movie findById(@PathVariable("id") int id) {
+        return movieService.findById(id);
     }
 
-    @PostMapping("/directorList")
+    @GetMapping("/movies")
+    public List<Movie> findByName(@RequestParam String movieName) {
+        return movieService.findByName(movieName);

```

Comment:
`/movies?movieName="映画の名前"&director="監督の名前"`
でAND検索できるような API を作りましょう。

Controllerの引数を変更してください。
それから、Serviceにも検索用のメソッドを作ってください。
メソッド名は findBy としましょう。
そして、映画の名前と監督の名前を引数に取れるようにしましょう。
```suggestion
    @GetMapping("/movies")
    public List<Movie> findByName(@RequestParam String movieName, @RequestParam String director) {
        return movieService.findBy(movieName, director);

```

検索処理作成は、わかりそうならそのまま実装いただいてよいですが、分からなければここまで実装してみてください。

Reply:
ひとまず、`/movies?movieName="映画の名前"`のようにmovieNameのみの場合は考慮しなくてよいです。
また、検索結果がない場合、クエリ文字列での検索は 200 で空の配列を返すようにしましょう。

Reply:
映画名と監督名のAND検索するAPIを作成しました。

## コミットハッシュ
d80aff17e4162eafd3ead8d2aed4d4782863c1d4

Reply:
確認します。

Reply:
対応ありがとうございます。
では、次のような実装に変更してみてください。

Reply:
参考
https://github.com/yukareto/Java-final-assignment/pull/6#discussion_r1451641140

Reply:
findByメソッドを変更しました。

コミットハッシュ
0f14a4c26d84782f0fb678955faf72072f7df31d

Reply:
いいとおもいます！
そうするとこちら消せますね。
https://github.com/ukrage/Assignment9/blob/0f14a4c26d84782f0fb678955faf72072f7df31d/src/main/java/com/uk/mybatisdemo202401/MovieController.java#L15-L18

Reply:
findAllメソッドを削除しました。
58bd80d93cde5ce4cba2b05bb289fc1a4af8fefc

Reply:
OKです。